Assets/Scripts/PlayerHealth.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class PlayerHealth : MonoBehaviour
+{
+    [SerializeField]
+    private float health = 100f;
+    [SerializeField]
+    private float fireDamage = 10f;
+    [SerializeField]
+    private float burnReactivationTime = 0.5f;
+
+    public bool isCheckingBurn = false;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (!isCheckingBurn)
+        {
+            if (IsTileOnFire())
+            {
```

Comment:
Why do you never use one if sentence with both checks? :D 
```suggestion
        if (!isCheckingBurn && IsTileOnFire())
        {
```